fps/src/main/scala/chapter03/exercise3_1.scala

```diff
@@ -0,0 +1,38 @@
+package chapter03
+
+import chapter03.exercise3_1.List.sum
+
+//(3, 14, 12,9,101,42
+object exercise3_1 extends App {
+  sealed trait List[+A]
+  case object Nil extends List[Nothing]
+  case class Cons[+A](head: A, tail: List[A]) extends List[A]
+
+  object List {
+    def sum(ints: List[Int]): Int = ints match {
+      case Nil => 0
+      case Cons(x, xs) => x + sum(xs)
+    }
+
+    def product(ds: List[Double]): Double = ds match {
+      case Nil => 1.0
+      case Cons(0.0, _) => 0.0
+      case Cons(x, xs) => x * product(xs)
+    }
+
+    def apply[A](as: A*): List[A] =
+      if (as.isEmpty) Nil
+      else Cons(as.head, apply(as.tail: _*))
```

Comment:
fyi - scala 3 syntax calling apply is `apply(as.tail*)`, ie. mirroring the `as: A*` type definition syntax

Reply:
fyi